api/v3/Payment.php

```diff
@@ -98,21 +98,26 @@ function civicrm_api3_payment_delete(&$params) {
  * @return array
  *   Api result array
  */
-function civicrm_api3_payment_cancel(&$params) {
+function civicrm_api3_payment_cancel($params) {
   $eftParams = array(
     'entity_table' => 'civicrm_contribution',
     'financial_trxn_id' => $params['id'],
   );
   $entity = civicrm_api3('EntityFinancialTrxn', 'getsingle', $eftParams);
-  $contributionId = $entity['entity_id'];
-  $params['total_amount'] = $entity['amount'];
-  unset($params['id']);
 
-  $trxn = CRM_Contribute_BAO_Contribution::recordAdditionalPayment($contributionId, $params, 'refund', NULL, FALSE);
```

Comment:
@eileenmcnaughton I think this API should be called for cancelled contribution, isn't it? Because the refund handling here isn't quite right.

Reply:
@monishdeb right - this PR can't be merged until we've resolved the refund handling - which we could add to this PR - I think just extracting makes sense - shall I add that so you can see what I mean?

Reply:
Sure why not :)